charts/harvester-cloud-provider/dependency_charts/kube-vip/templates/daemonset.yaml

```diff
@@ -0,0 +1,51 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: {{ include "kube-vip.name" . }}
+  namespace: {{ .Release.Namespace | default "kube-system" }}
+spec:
+  selector:
+    matchLabels:
+      {{- include "kube-vip.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      labels:
+        {{- include "kube-vip.selectorLabels" . | nindent 8 }}
+    spec:
+      containers:
+      - args:
+          - manager
+        env:
+          {{- if eq .Values.env.cp_enable "true" }}
+          - name: vip_address
+            value: {{ required "A valid config.address required!" .Values.config.address}}
+          {{- end }}
+        {{- with .Values.env }}
+          {{- range $k, $v := . }}
+            {{- $name := $k }}
+            {{- $value := $v }}
+          - name: {{ quote $name }}
+            value: {{ quote $value }}
+          {{- end }}
+        {{- end }}
+        image: {{ template "system_default_registry" . }}{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}
```

Comment:
need to define the template per https://github.com/rancher/rke2-charts/blob/main-source/packages/rke2-canal/charts/templates/_helpers.tpl#L1